### PR TITLE
check that missing default rules are not synced while syncing rules on a switch

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 1]
```diff
@@ -214,6 +215,33 @@ class SwitchSyncSpec extends BaseSpecification {
...
+
+        when: "Synchronize rules on the switch"
+        northbound.synchronizeSwitchRules(sw.dpId)
+
+        then: "Missing rules are not synchronized"
+        def rulesAfterSync = northbound.validateSwitchRules(sw.dpId)
+        amountProperRules == rulesAfterSync.properRules.size()
+        amountMissingRules == rulesAfterSync.missingRules.size()
+        amountExcessRules == rulesAfterSync.excessRules.size()
+
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: Why `defaultRules`, `amountProperRules`, `amountMissingRules`, `amountExcessRules`, `initRules` are required in this test?
just do
```
then: "Missing rules are not synchronized"
with(northbound.validateSwitchRules(sw.dpId)) {
   properRules.empty
   excessRules.empty
   missingRules.sort() == sw.defaultCookies.sort()
}
```
that's it

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 2]
```diff
@@ -214,6 +215,33 @@ class SwitchSyncSpec extends BaseSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    def "System doesn't synchronize missing default rules"() {
+        given: "An active switch with missing rules"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: missing _default_ rules

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 3]
```diff
@@ -214,6 +215,33 @@ class SwitchSyncSpec extends BaseSpecification {
...
+        def initRules = northbound.validateSwitchRules(sw.dpId)
+        def amountProperRules = initRules.properRules.size()
+        def amountMissingRules = initRules.missingRules.size()
+        assert defaultRules.size() == amountMissingRules
+        def amountExcessRules = initRules.excessRules.size()
+
+        when: "Synchronize rules on the switch"
+        northbound.synchronizeSwitchRules(sw.dpId)
+
+        then: "Missing rules are not synchronized"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: _default_ rules

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: ok, fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: thx, fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 4]
```diff
@@ -214,6 +215,29 @@ class SwitchSyncSpec extends BaseSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    def "System doesn't synchronize missing default rules"() {
```
Comment (dpoltavets) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: The presence of missing default rules will not affect anything. But the presence of excess default rules will lead to the fact that during synchronization, along with excess default rules, the proper default rules are deleted.

Comment (dpoltavets) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: The presence of missing default rules will not affect anything. But the presence of excess default rules will lead to the fact that during synchronization, along with excess default rules, the proper default rules are deleted.

**Comment by andriidovhan** [comment]
test will be implemented when new module for FL is created
new module should allow create any rule what we want

**Pull request closed without merging**